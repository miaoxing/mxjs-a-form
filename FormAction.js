import { CListBtn } from '@mxjs/a-clink';
import { Button, Form, Space } from 'antd';
import PropType from 'prop-types';
import useForm from './useForm';

const FormAction = ({url, list = true, ...props}) => {
  const {loading} = useForm();

  return (
    <Form.Item
      wrapperCol={{offset: 4, span: 8}}
      {...props}
    >
      <Space>
        <Button htmlType="submit" type="primary" loading={loading}>
          提交
        </Button>
        {list && (url ? <Button href={url}>返回列表</Button> : <CListBtn/>)}
      </Space>
    </Form.Item>
  );
};

FormAction.propTypes = {
  url: PropType.string,
  list: PropType.bool,
};

export default FormAction;
