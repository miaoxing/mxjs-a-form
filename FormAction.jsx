import { CListBtn } from '@mxjs/a-clink';
import { Button, Form, Space } from 'antd';
import PropType from 'prop-types';
import useForm from './useForm';
import { Fragment } from 'react';
import { Section } from '@mxjs/a-section';

const FormAction = ({url, list = true, variant = 'card', ...props}) => {
  const {loading} = useForm();
  const Container = variant === 'card' ? Section : Fragment;

  return (
    <Container>
      <Form.Item
        wrapperCol={{offset: 4, span: 8}}
        {...props}
      >
        <Space>
          <Button htmlType="submit" type="primary" loading={loading}>
            提交
          </Button>
          {list && (url ? <Button href={url}>返回列表</Button> : <CListBtn/>)}
        </Space>
      </Form.Item>
    </Container>
  );
};

FormAction.propTypes = {
  url: PropType.string,
  list: PropType.bool,
  variant: PropType.oneOf(['card', 'none']),
};

export default FormAction;
