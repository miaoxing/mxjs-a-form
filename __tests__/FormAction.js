import FormAction from '../FormAction';
import {MemoryRouter} from 'react-router';
import {render} from '@testing-library/react';

describe('FormAction', () => {
  test('render', async () => {
    const {container} = render(<MemoryRouter><FormAction/></MemoryRouter>);

    expect(container).toMatchSnapshot();
  });

  test('custom offset', async () => {
    const {container} = render(<MemoryRouter><FormAction wrapperCol={{offset: 8}}/></MemoryRouter>);

    expect(container).toMatchSnapshot();
  });
});
