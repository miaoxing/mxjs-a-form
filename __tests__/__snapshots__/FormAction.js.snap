// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormAction custom offset 1`] = `
<div>
  <div
    class="ant-form-item"
  >
    <div
      class="ant-row ant-form-item-row"
    >
      <div
        class="ant-col ant-col-offset-8 ant-form-item-control"
      >
        <div
          class="ant-form-item-control-input"
        >
          <div
            class="ant-form-item-control-input-content"
          >
            <div
              class="ant-space ant-space-horizontal ant-space-align-center"
            >
              <div
                class="ant-space-item"
                style="margin-right: 8px;"
              >
                <button
                  class="ant-btn ant-btn-primary"
                  type="submit"
                >
                  <span>
                    提 交
                  </span>
                </button>
              </div>
              <div
                class="ant-space-item"
              >
                <a
                  class="ant-btn ant-btn-default"
                  href="/"
                >
                  <span>
                    返回列表
                  </span>
                </a>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`FormAction render 1`] = `
<div>
  <div
    class="ant-form-item"
  >
    <div
      class="ant-row ant-form-item-row"
    >
      <div
        class="ant-col ant-col-8 ant-col-offset-4 ant-form-item-control"
      >
        <div
          class="ant-form-item-control-input"
        >
          <div
            class="ant-form-item-control-input-content"
          >
            <div
              class="ant-space ant-space-horizontal ant-space-align-center"
            >
              <div
                class="ant-space-item"
                style="margin-right: 8px;"
              >
                <button
                  class="ant-btn ant-btn-primary"
                  type="submit"
                >
                  <span>
                    提 交
                  </span>
                </button>
              </div>
              <div
                class="ant-space-item"
              >
                <a
                  class="ant-btn ant-btn-default"
                  href="/"
                >
                  <span>
                    返回列表
                  </span>
                </a>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
